### ISSUE TITLE
Can't enable module after installation

### ISSUE THREAD

**Issue by x3a** [felixhahnweilheim/humhub-dark-mode#48]
I'm running HumHub 1.15.2 and can't enable the module after installation.

First, I had dark mode 1.0.4 installed and couldn't enable it, there was no debugging output in the HumHub logs.

Today I did an update to dark mode 1.0.5 and this appeared in the logs during the process:

```
yii\base\ErrorException: Attempt to read property "settings" on null in /***/protected/modules/dark-mode/models/Config.php:25
Stack trace:
#0 /***/protected/modules/dark-mode/models/Config.php(25): yii\base\ErrorHandler->handleError()
#1 /***/protected/vendor/yiisoft/yii2/base/BaseObject.php(109): humhub\modules\darkMode\models\Config->init()
#2 /***/protected/modules/dark-mode/assets/DarkStyleAsset.php(40): yii\base\BaseObject->__construct()
#3 /***/protected/vendor/yiisoft/yii2/base/BaseObject.php(109): humhub\modules\darkMode\assets\DarkStyleAsset->init()
#4 /***/protected/humhub/components/Module.php(183): yii\base\BaseObject->__construct()
#5 /***/protected/humhub/modules/marketplace/services/ModuleService.php(105): humhub\components\Module->publishAssets()
#6 /***/protected/humhub/modules/marketplace/controllers/UpdateController.php(48): humhub\modules\marketplace\services\ModuleService->update()
#7 [internal function]: humhub\modules\marketplace\controllers\UpdateController->actionInstall()
#8 /***/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#9 /***/protected/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
#10 /***/protected/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()
#11 /***/protected/vendor/yiisoft/yii2/web/Application.php(103): yii\base\Module->runAction()
#12 /***/protected/vendor/yiisoft/yii2/base/Application.php(384): yii\web\Application->handleRequest()
#13 /***/index.php(25): yii\base\Application->run()
#14 {main} 
```
I still can't enable the module after it's installation and there's no additional debugging output in the HumHub logs after I hit "enable".

Since this is very little information to properly debug this, let me know if there's something I can do to help resolving this.

**Comment by felixhahnweilheim** [comment]
It's actually a bit hard to understand the error because I don't know how to reproduce it.

`Yii::$app->getModule('dark-mode')` returns null. So maybe it is called too early in the

**Comment by felixhahnweilheim** [comment]
Now I was able to reproduce the error log:

- Module is installed in v1.0.4 but not active.
- Update to 1.0.5.
- The error log you posted above appears.
However the module is properly updated and I can enable it without problems.

(I'll look into fixing this minor bug later.)
The more important question is why you can't enable the module.

Have you tried via command line? Sometimes that gives better feedback.

I have now tried everything with APC(u) caching. Not sure if another caching could be an issue.

Also file permissions could be an issue but here I am really just guessing.

**Comment by x3a** [comment]
Alright, thank you for the useful feedback, I tested some things and can provide more information now:

My test instance initially had the Option `Cache Backend` set to `File`.

I tried activating the module via the CLI, which returned no errors and claimed the operation was successful, but it was still listed as deactivated after this in the admin web panel.

Finally, I changed `Cache Backend` to `APC(u)` and was able to activate the module successfully via the admin web panel.
